statusHandler/termuic/termuiRenders/widgetsRender.go

```diff
@@ -27,14 +27,14 @@ type WidgetsRender struct {
 	networkRecv *widgets.Gauge
 	networkSent *widgets.Gauge
 
-	termuiConsoleMetrics *sync.Map
+	presenter statusHandler.PresenterInterface
 }
 
 //NewWidgetsRender method will create new WidgetsRender that display termui console
-func NewWidgetsRender(metricData *sync.Map, grid *DrawableContainer) *WidgetsRender {
+func NewWidgetsRender(presenter statusHandler.PresenterInterface, grid *DrawableContainer) *WidgetsRender {
 	self := &WidgetsRender{
```

Comment:
should add checks for nil and should add tests for those checks

Reply:
Checks for nil was added for constructor parameters and if an parameter is nil an error will be returned.